include/download.h

```diff
@@ -23,6 +25,8 @@ class Download {
 public:
 	explicit Download(std::function<void()> cb_require_view_update);
 	~Download();
+	Download(const Download& download);
+	Download& operator=(const Download& download);
```

Comment:
Why are these needed?

Reply:
Because after migrating from `std::string` to `Filepath` class `Download` now got a member of type `Filepath`.

https://github.com/newsboat/newsboat/blob/201a939e8a33e7f56df9e34f94737bce8e04ad8f/include/download.h#L61

Default copy ctor/optor would call a `Filepath` default ctor/optor, but they were marked as deleted. As consequence compile error.

Reply:
Please don't mark comments resolved. I think it's *my* right to decide if my comment was addressed or not ;)

I understand the rationale now, but I think the solution is quite brittle. If we ever add another field to `Download`, we'd have to remember to update these constructor and operator too. Chances are that we won't, and then boom!—we have a bug.

I'd prefer to add a copy constructor and operator to `Filepath` itself. That would also spare us from all these `clone()` calls which look fine in Rust but are quite outlandish in C++. What do you think?

Reply:
> Please don't mark comments resolved. I think it's my right to decide if my comment was addressed or not ;)
Ok, since now I will be leave a commit hash in such cases.

Reply:
`Box` is indeed uncopiable, but you can ask Rust to clone it—see how `Filepath::clone()` is implemented. I think you can just copy the code from there and it will work. Once you got a copy constructor, `clone()` can be simplified to something like `return Filepath(*this)`. Can you please take another stab at this? Please push it even if it doesn't work, we'll figure it out.

Reply:
Done! https://github.com/newsboat/newsboat/pull/2561/commits/aef5955cf2d373b8e5e17e4a768ed7a5d73dc156

Reply:
Excellent!